lib/graphql/client/http.rb

```diff
@@ -64,11 +64,21 @@ def execute(document:, operation_name: nil, variables: {}, context: {})
         request["Content-Type"] = "application/json"
         headers(context).each { |name, value| request[name] = value }
 
+        form_fields = form_data!(variables)
         body = {}
         body["query"] = document.to_query_string
         body["variables"] = variables if variables.any?
         body["operationName"] = operation_name if operation_name
-        request.body = JSON.generate(body)
+
+        if form_fields
+          # post as multipart/form-data to stream file contents
+          form_fields.append(["operations", JSON.generate(body)])
+          request.set_form(form_fields, "multipart/form-data")
```

Comment:
GraphQL [Multipart form field structure](https://github.com/jaydenseric/graphql-multipart-request-spec#multipart-form-field-structure) specifies fields order, `operations`, `map`, File fields.
But elements of `form_fields` is ordered, File fields, `map`, `operations`.

```suggestion
          form_fields.prepend(["operations", JSON.generate(body)])
          request.set_form(form_fields, "multipart/form-data")
```

Reply:
For example, the GrarpQL server implementation of [gqlgen](https://github.com/99designs/gqlgen) checks fields order strictly.